### PR TITLE
fix build without mpi

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

**Comment by dalg24** [comment]
Please describe the rational for these changes.  Did you actually encounter places where we compile this file and MPI is not enabled?

File: core/unit_test/CMakeLists.txt [hunk 1]
```diff
@@ -77,4 +77,6 @@ macro(Cabana_add_tests)
 endmacro()
 
 Cabana_add_tests(NAMES AoSoA Slice DeepCopy Tuple Sort LinkedCellList NeighborList Parallel)
-Cabana_add_tests(MPI NAMES CommunicationPlan Distributor Halo)
+if(Cabana_ENABLE_MPI)
```
Comment (dalg24) [inline core/unit_test/CMakeLists.txt]: OK that's the correct fix.

Comment (junghans) [reply core/unit_test/CMakeLists.txt]: Yeah, the first fix wasn't more a workaround.

**Comment by dalg24** [comment]
> Please describe the rational for these changes. Did you actually encounter places where we compile this file and MPI is not enabled?

For the record this comment was for a different version that introduced preprocessor directives in `mpi_unit_test_main.cpp`.

**Comment by junghans** [comment]
> > Please describe the rational for these changes. Did you actually encounter places where we compile this file and MPI is not enabled?
> 
> For the record this comment was for a different version that introduced preprocessor directives in `mpi_unit_test_main.cpp`.

https://github.com/ECP-copa/Cabana/commit/cb05569174a3f7e14e5ae714d6ced4f45d5c7b0b

**Pull request merged**